opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java

```diff
@@ -1122,53 +1132,43 @@ private static String readOpening(InputStream in, byte[] sig)
         }
 
         int nRead = 0;
-        boolean sawNonWhitespace = false;
-        boolean lastWhitespace = false;
-        boolean postHashbang = false;
+        boolean ignoreWhiteSpace = true;
+        boolean breakOnNewLine = false;
         int r;
 
         StringBuilder opening = new StringBuilder();
         BufferedReader readr = new BufferedReader(new InputStreamReader(in, encoding), OPENING_MAX_CHARS);
+
         while ((r = readr.read()) != -1) {
-            if (++nRead > OPENING_MAX_CHARS) {
```

Comment:
I'd swear that somewhere I saw a `Reader` that is limited by the amount of characters it can read. It would work well in this case, I think.

Reply:
I was not able to find a reader with max char limit, changed the loop to for with exit condition based upon characters read.